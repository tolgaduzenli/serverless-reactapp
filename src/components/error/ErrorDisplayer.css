.error-container{
    display: flex;
    justify-content: center;
    align-items: center;
    top: 50px;
    background-color: #ef5350;
}