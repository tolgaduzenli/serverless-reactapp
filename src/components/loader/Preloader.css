.preloader-container{
    display:block;
    top:150px;
}