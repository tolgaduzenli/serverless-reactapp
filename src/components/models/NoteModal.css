.modal-view {
    display: flex;
    flex-direction:column;
    justify-content: center;
    align-items: space-between;
}

.modal-content{
    display:flex;
    flex-direction:column;
    margin: 15px;
}