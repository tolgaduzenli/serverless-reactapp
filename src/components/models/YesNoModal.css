.modal-view {
    display: flex;
    flex-direction:column;
    justify-content: center;
    align-items: space-between;
}

.button-group{
    display:flex;
    flex-direction:row;
    justify-content: center;
}

.button-group > button{
    margin:15px;
}