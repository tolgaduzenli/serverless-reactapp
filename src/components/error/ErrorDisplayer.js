import React from 'react'
import './ErrorDisplayer.css'

const ErrorDisplayer = () => {
    return <div className="error-container">
        <div>
            <p>Error has been occurred.Please try again later</p>
        </div>
    </div>
}

export default ErrorDisplayer