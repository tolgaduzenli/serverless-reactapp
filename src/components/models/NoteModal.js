import React from 'react'
import Modal from 'react-awesome-modal'
import './NoteModal.css'
class NoteModal extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            newDescription: this.props.note.description,
            newTitle: this.props.note.title,
            _id: this.props.note._id
        }
    }
    onTitleChange = (e) => {
        this.setState({ newTitle: e.target.value })
    }
    onDescriptionChange = (e) => {
        this.setState({ newDescription: e.target.value })
    }
    render() {
        const { closeModal, showModal, onClickModalSave } = this.props
        const { newTitle, newDescription, _id } = this.state
        return <section>
            <Modal
                visible={showModal}
                width="400"
                height="200"
                effect="fadeInUp"
                onClickAway={() => closeModal()}
            >
                <div className="modal-view">
                    <form className="modal-content">
                        <input type="text"
                            placeholder="title"
                            defaultValue={newTitle}
                            onChange={this.onTitleChange} />
                        <input type="text"
                            placeholder="description"
                            defaultValue={newDescription}
                            onChange={this.onDescriptionChange} />
                    </form>
                    <div>
                        <button className="waves-effect waves-light btn-small blue lighten-2"
                            onClick={() => onClickModalSave(newTitle, newDescription, _id)}>Save
                    </button>
                    </div>
                </div>
            </Modal>
        </section>
    }
}

export default NoteModal
