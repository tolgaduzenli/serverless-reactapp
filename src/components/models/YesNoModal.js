import React from 'react'
import Modal from 'react-awesome-modal'
import './YesNoModal.css'

class YesNoModal extends React.Component {

    render() {
        const { onClickYes, onClickNo, note, showModal } = this.props
        return <section>
            <Modal
                visible={showModal}
                width="400"
                height="200"
                effect="fadeInUp"
                onClickAway={() => onClickNo()}
            >
                <div className="modal-view">
                    <div>Do you want to remove the note</div>
                    <div>Title: {note.title}</div>
                    <div>Description: {note.description}</div>
                    <div className="button-group">
                        <button className="waves-effect waves-light btn-small green lighten-1"
                            onClick={() => onClickYes(note._id)}>Yes
                        </button>
                        <button className="waves-effect waves-light btn-small grey"
                            onClick={() => onClickNo()}>No
                        </button>
                    </div>
                </div>
            </Modal>
        </section>
    }
}

export default YesNoModal
