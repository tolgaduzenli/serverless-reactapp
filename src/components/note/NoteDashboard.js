import React from 'react'
import NoteList from './NoteList'
import NoteModal from '../models/NoteModal'
import Preloader from '../loader/Preloader';
import ErrorDisplayer from '../error/ErrorDisplayer'
import YesNoModal from '../models/YesNoModal'
import { getAllNotesFromService, saveNewNote, updateNote, deleteNote } from '../../service/APICall'

class NoteDashboard extends React.Component {
    constructor() {
        super()
        this.state = {
            noteList: null,
            showModal: false,
            showYesNoModal: false,
            note: {
                title: "",
                description: ""
            },
            loading: false,
            error: false
        }
    }
    componentDidMount() {
        this.getAllNotes()
    }

    getAllNotes = () => {
        this.setState({ loading: true })
        getAllNotesFromService().then(result => {
            this.setState({
                noteList: result.data,
                loading: false,
                error: false
            })
        }).catch(err => {
            console.log(err)
            this.setState({ loading: false, error: true })
        })
    }

    onClicAddButton = () => {
        const emptyNote = {
            title: "",
            description: ""
        }
        this.setState({ note: emptyNote })
        this.toggleModal()
    }

    onClickEditButton = (selectedNote) => {
        this.setState({ note: selectedNote })
        this.toggleModal()
    }

    onClickDeleteButton = (selectedNote) => {
        this.setState({ note: selectedNote })
        this.toggleYesNoModal()
    }

    onClickDeleteYes = (id) => {
        this.toggleYesNoModal()
        this.setState({ loading: true })
        deleteNote(id).then(() => {
            this.getAllNotes()
        }).catch(err => {
            this.setState({ error: true, loading: false })
            console.log(err)
        })
    }

    toggleModal = () => {
        this.setState({ showModal: !this.state.showModal });
    }

    toggleYesNoModal = () => {
        this.setState({ showYesNoModal: !this.state.showYesNoModal });
    }

    onClickModalSave = (newTitle, newDescription, _id) => {
        this.setState({ loading: true })
        this.toggleModal()
        if (_id === undefined || _id === null) {
            saveNewNote(newTitle, newDescription).then(result => {
                this.getAllNotes()
            }).catch(err => {
                console.log(err)
                this.setState({ error: true, loading: false })
            })
        } else {
            updateNote(_id, newTitle, newDescription).then(result => {
                this.getAllNotes()
            }).catch(err => {
                console.log(err)
                this.setState({ error: true, loading: false })
            })
        }
    }

    renderPreLoader = () => {
        return <Preloader />
    }

    renderDashboard = () => {
        const { noteList, showModal, note, error, showYesNoModal } = this.state
        return <div>
            {error && <ErrorDisplayer />}
            <NoteList noteList={noteList}
                onClickAdd={this.onClicAddButton}
                onClickDelete={this.onClickDeleteButton}
                onClickEdit={this.onClickEditButton} />
            {showModal && <NoteModal closeModal={this.toggleModal}
                showModal={showModal}
                onClickModalSave={this.onClickModalSave}
                note={note}
            />}
            {showYesNoModal && <YesNoModal onClickNo={this.toggleYesNoModal}
                showModal={showYesNoModal}
                onClickYes={this.onClickDeleteYes}
                note={note} />}
        </div>
    }

    render() {
        if (this.state.loading) {
            return this.renderPreLoader()
        } else {
            return this.renderDashboard()
        }
    }
}

export default NoteDashboard