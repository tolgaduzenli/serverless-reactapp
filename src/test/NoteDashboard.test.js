import React from 'react'
import enzyme from 'enzyme'
import NoteDashboard from '../components/note/NoteDashboard'
import NoteList from '../components/note/NoteList'
import Preloader from '../components/loader/Preloader'
import ErrorDisplayer from '../components/error/ErrorDisplayer'
import NoteModal from '../components/models/NoteModal'
import YesNoModal from '../components/models/YesNoModal'

describe('<NoteDashboard />', () => {
    let wrapper
    beforeEach(() => {
        wrapper = enzyme.shallow(
            <NoteDashboard />
        )
        wrapper.setState({ loading: false })
    })

    it('should render correctly', () => expect(wrapper).toMatchSnapshot())

    it('should render a <div />', () => {
        expect(wrapper.find('div').length).toEqual(1)
    })

    it('should render a <NoteList /> ', () => {
        expect(wrapper.containsMatchingElement(
            <NoteList noteList={wrapper.instance().noteList}
                onClickAdd={wrapper.instance().onClickAdd}
                onClickDelete={wrapper.instance().onClickDelete}
                onClickEdit={wrapper.instance().onClickEdit} />
        )).toEqual(true)
    })

    it('should render a <ErrorDisplayer /> ', () => {
        wrapper.setState({ error: true })
        expect(wrapper.containsMatchingElement(
            <ErrorDisplayer />
        )).toEqual(true)
    })

    it('should render a <NoteModal /> ', () => {
        wrapper.setState({ showModal: true })
        expect(wrapper.containsMatchingElement(
            <NoteModal />
        )).toEqual(true)
    })

    it('should render a <YesNoModal /> ', () => {
        wrapper.setState({ showYesNoModal: true })
        expect(wrapper.containsMatchingElement(
            <YesNoModal />
        )).toEqual(true)
    })

    describe('while "loading" is true', () => {
        let wrapper
        beforeEach(() => {
            wrapper = enzyme.shallow(
                <NoteDashboard />
            )
            wrapper.setState({ loading: true })
        })

        it('should render a <Preloader />', () => {
            expect(wrapper.containsMatchingElement(
                <Preloader />
            )).toEqual(true)
        })
    })
})