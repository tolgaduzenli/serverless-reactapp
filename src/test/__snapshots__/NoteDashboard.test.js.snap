// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoteDashboard /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoteDashboard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <NoteList
          noteList={null}
          onClickAdd={[Function]}
          onClickDelete={[Function]}
          onClickEdit={[Function]}
        />,
        false,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "noteList": null,
          "onClickAdd": [Function],
          "onClickDelete": [Function],
          "onClickEdit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <NoteList
            noteList={null}
            onClickAdd={[Function]}
            onClickDelete={[Function]}
            onClickEdit={[Function]}
          />,
          false,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "noteList": null,
            "onClickAdd": [Function],
            "onClickDelete": [Function],
            "onClickEdit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
